pkg/testrunner/result/status-uploader.go

```diff
@@ -179,23 +406,86 @@ func getGithubClient(repoURL *url.URL, githubUser, githubPassword string) (*gith
 	}
 	githubClient, err := util.GetGitHubClient(apiURL, githubUser, githubPassword, uploadURL, true)
 	if err != nil {
-		return nil, err
+		l.Error(err, fmt.Sprintf("failed to get github client for %s", component))
+		return nil
 	}
-	return githubClient, nil
+	return githubClient
 }
 
 // MarkTestrunsAsIngested sets the ingest status of testruns to true
-func MarkTestrunsAsUploadedToGithub(log logr.Logger, tmClient kubernetes.Interface, tr *tmv1beta1.Testrun) error {
+func MarkTestrunsAsUploadedToGithub(log logr.Logger, tmClient kubernetes.Interface, runs *testrunner.RunList) error {
 	ctx := context.Background()
 	defer ctx.Done()
 
-	enabled := true
-	tr.Status.UploadedToGithub = &enabled
-	err := tmClient.Client().Update(ctx, tr)
+	for _, run := range *runs {
+		tr := run.Testrun
+		enabled := true
+		tr.Status.UploadedToGithub = &enabled
+		err := tmClient.Client().Update(ctx, tr)
+		if err != nil {
+			log.Error(err, fmt.Sprintf("unable to update status of testrun %s in namespace: %s", tr.Name, tr.Namespace))
+			return err
+		}
+	}
+	log.Info("successfully updated status of testrun")
+	return nil
+}
+
+type ComponentExtended struct {
+	PlainURL        string
+	Version         string
+	GithubClient    *github.Client
+	Owner           string
+	Name            string
+	GithubReleaseID int64
+}
+
+func parseComponent(component *componentdescriptor.Component, githubUser string, githubPassword string) (ComponentExtended, error) {
```

Comment:
some documentation what the function is doing and what a extended component is

Reply:
Will add documentation: wraps component struct with additional github properties: github client, repo owner, repo name, release ID.